src/lib/Form/Processor/ContentFormProcessor.php

```diff
@@ -124,14 +124,24 @@ public function processSaveDraftAndClose(FormActionEvent $event): void
         $draft = $this->saveDraft($data, $languageCode, []);
         $referrerLocation = $event->getOption('referrerLocation');
 
+        if ($referrerLocation === null) {
+            $versionInfo = $data->contentDraft->getVersionInfo();
```

Comment:
I think there may be an additional case where `$referrerLocation` is not set (it's optional) but you are saving draft of already published Content Item and you should be redirected back to its location instead of parent.